app/src/androidTest/java/com/github/h3lp3rs/h3lp/HelpeeSelectionActivityTest.kt

```diff
@@ -0,0 +1,212 @@
+    fun clickPhoneButtonDialsCorrectEmergencyNumber() {
+        mockLocationToCoordinates(SWISS_LONG, SWISS_LAT)
+        loadValidMedicalDataToStorage()
+
+        launchAndDo {
+            // Clicking on the call for emergency button
+            val phoneButton = onView(withId(R.id.help_params_call_button))
+            phoneButton.perform(click())
+
+            // click the ambulance in the popup
+            onView(withId(R.id.ambulance_call_button)).inRoot(RootMatchers.isFocusable())
+                .perform(click())
+
+            // The expected ambulance phone number given the location (specified by the coordinates)
+            // val number = "tel:${SWISS_EMERGENCY_NUMBER}"
+
+            // Checking that this emergency number is dialed
+            intended(
+                allOf(
+                    hasAction(ACTION_DIAL) //,
+                    // hasData(Uri.parse(number)) Cirrus doesn't like this
+                )
+            )
+        }
+    }
+
+    @Test
+    fun clickPhoneButtonWithNoLocationDialsDefaultEmergencyNumber() {
+        loadValidMedicalDataToStorage()
+        mockFailingLocation()
+
+        launchAndDo {
+            val phoneButton = onView(withId(R.id.help_params_call_button))
+            phoneButton.perform(click())
+
+            // click the ambulance in the popup
+            onView(withId(R.id.ambulance_call_button)).perform(click())
+
+            intended(
+                allOf(
+                    hasAction(ACTION_DIAL)
+                    // Not cheking actual number because of Cirrus
+                )
+            )
+        }
+    }
+
+    @Test
+    fun clickSearchHelpWithNoMedsDoesNotChangeActivity() {
+        launchAndDo {
+            val searchHelpButton = onView(withId(R.id.help_params_search_button))
+            searchHelpButton.check(matches(isDisplayed()))
+            searchHelpButton.perform(click())
+            // No new intent:
+            assertThat(getIntents().size, `is`(0))
+        }
+    }
+
+
+    @Test
+    fun screenDisplaysCorrectLocation() {
+        mockLocationToCoordinates(SWISS_LONG, SWISS_LAT)
+
+        launchAndDo {
+            // Checking that the user's actual location is displayed before they call an ambulance
+            val locationInformation = onView(withId(R.id.location_information))
+            locationInformation
+                .check(matches(withText(containsString(SWISS_LONG.toString()))))
+            locationInformation
+                .check(matches(withText(containsString(SWISS_LAT.toString()))))
+        }
+    }
+
+    private fun launch(): ActivityScenario<MainPageActivity> {
+        return ActivityScenario.launch(
+            Intent(
+                getApplicationContext(),
+                HelpeeSelectionActivity::class.java
+            )
+        )
+    }
+
+    private fun launchAndDo(action: () -> Unit) {
+        launch().use {
+            initIntentAndCheckResponse()
+            action()
+            release()
```

Comment:
I guess that for consistency, these could be at the top of the file after the setup method as that's where they usually are in other test methods